### PR TITLE
DAOS-4456 control: organise server instance files

### PR DESCRIPTION
**Pull request by tanabarr**
In preparation for adding capability to manage IO server instances
independently, subdivide instance functionality between files.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/control/server/instance_exec.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+	"github.com/pkg/errors"
+
+	srvpb "github.com/daos-stack/daos/src/control/common/proto/srv"
+	"github.com/daos-stack/daos/src/control/server/ioserver"
+)
+
+// IOServerRunner defines an interface for starting and stopping the
+// daos_io_server.
+type IOServerRunner interface {
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/control/server/instance_exec.go:38]: wanted a name that covered routines that would handle execution of the instance e.g. Start() processing loop in addition to the underlying process runner. will think about it more in the subsequent PR. @mjmac have a preference?

[the remark this answers is not in the document]
Comment (mjmac) [reply src/control/server/instance_exec.go:38]: I don't feel strongly either way. Logically, if we're following a pattern of instance_$noun-describing-contents, then exec probably fits better than runner.

[the remark this answers is not in the document]
Comment (kjacque) [reply src/control/server/instance_exec.go:38]: I'm fine as-is. Let's land it.

**Comment by tanabarr** [comment]
intermittent failures in CI:
NvmeObject https://jira.hpdd.intel.com/browse/DAOS-4567
IorSmall "Test interrupted by SIGTERM" during pool connect ???
MdtestSmall ???

**Pull request merged**